main.tf

```
@@ -92,3 +100,57 @@ resource "kubernetes_namespace" "namespace" {
     labels = each.value.labels
   }
 }
+
+resource "kubernetes_network_policy" "isolated" {
```

```suggestion
resource "kubernetes_network_policy_v1" "isolated" {
```

Use [`kubernetes_network_policy_v1`](https://registry.terraform.io/providers/hashicorp/kubernetes/latest/docs/resources/network_policy_v1) resource type.